src/VisualStudio/Core/Test.Next/Remote/SerializationValidator.cs

```diff
@@ -70,13 +71,13 @@ public ChecksumObjectCollection(SerializationValidator validator, WellKnownSynch
         }
 
         public SolutionAssetStorage AssetStorage { get; }
-        public ISerializerService Serializer { get; }
+        public TestSerializerService Serializer { get; }
         public HostWorkspaceServices Services { get; }
 
         public SerializationValidator(HostWorkspaceServices services)
         {
             AssetStorage = services.GetRequiredService<ISolutionAssetStorageProvider>().AssetStorage;
-            Serializer = services.GetRequiredService<ISerializerService>();
+            Serializer = (TestSerializerService)services.GetRequiredService<ISerializerService>();
```

Comment:
>services.GetRequiredService [](http://example.com/codeflow?start=48&length=27)

I assume it's the preferred design not to have an overload of Get*Service which takes in the desired implementation type and does the cast for the caller?

Reply:
yeah.  it's test code.  i don't have strong feeligns.